Pod/Classes/NJHNavigationBarBackgroundSelectionView.m

```diff
@@ -0,0 +1,173 @@
+ *  the center x of the selectorView. It is the constraint we manipulate to get the view to move right to left
+ */
+@property (weak, nonatomic) IBOutlet NSLayoutConstraint *animateableConstraint;
+
+/**
+ *  The number of sections this selection view should display, calculated from the labelTitles array
+ */
+@property (nonatomic) NSUInteger sections;
+
+/**
+ *  One for each section on the
+ */
+@property (nonatomic) NSMutableArray *labels;
+@end
+
+@implementation NJHNavigationBarBackgroundSelectionView
+
+/**
+ *  Since we are overriding both the setter and the getter for the following properties, we must synthesize them
+ */
+@synthesize fontForLabels = _fontForLabels;
+@synthesize colorForLabels = _colorForLabels;
+
++ (instancetype)instance {
+    NSBundle *resourceBundle = [NSBundle bundleWithPath:[[NSBundle mainBundle].bundlePath stringByAppendingString:kNJHFrameworkBundleName]];
+    UINib *nib = [UINib nibWithNibName:NSStringFromClass([self class]) bundle:resourceBundle];
+    return [[nib instantiateWithOwner:nil options:nil] firstObject];
+}
+
+- (void)didMoveToSuperview {
+    [super didMoveToSuperview];
+    
+    self.sections = [self.labelTitles count];
+    
+    // The constraint that determines the width of the selector view that floats on top of the background view to signify the current selection
+    NSLayoutConstraint *constraint = [NSLayoutConstraint constraintWithItem:self.selectorView
+                                                                  attribute:NSLayoutAttributeWidth
+                                                                  relatedBy:NSLayoutRelationEqual
+                                                                     toItem:self
+                                                                  attribute:NSLayoutAttributeWidth
+                                                                 multiplier:1.f / (CGFloat)self.sections
+                                                                   constant:0];
+    
+    constraint.priority = 999; // Let the system break this constraint when we are compressed against the sides of the background view
+    [self addConstraint:constraint];
+    
+    [self createLabelViews];
+    
+    [self addTarget:self action:@selector(viewTapped:event:) forControlEvents:UIControlEventTouchUpInside];
+    
+    self.layer.cornerRadius = CGRectGetHeight(self.frame) / 2;
+    self.selectorView.layer.cornerRadius = CGRectGetHeight(self.selectorView.frame) / 2;
+}
+
+- (void)createLabelViews {
+    [self.labelTitles enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
+        UILabel *label = [[UILabel alloc] init];
+        label.text = (NSString *)obj;
+        label.font = self.fontForLabels;
+        [label sizeToFit];
+        
+        [self.labels addObject:label];
+        [self addSubview:label];
```

Comment:
Could this potentially be called multiple times, thus increasingly add new labels?